src/mono/wasi/Wasi.Build.Tests/BuildThenRun.cs

```diff
@@ -13,113 +13,90 @@
+                                    .ExecuteWithCapturedOutput($"run --no-silent --no-build -c {config} x y z")
+                                    .EnsureSuccessful();
+
+        Assert.Contains("Hello, Wasi Console!", res.Output);
     }
 
-    [ConditionalTheory(typeof(BuildTestBase), nameof(IsUsingWorkloads))]
-    [MemberData(nameof(TestDataForConsolePublishAndRun))]
-    public void ConsolePublishAndRunForSingleFileBundle(string config, bool relinking, bool invariantTimezone)
+    [Theory]
+    [InlineData("Debug", /*aot*/ false, /*singleFileBundle*/ false)]
+    [InlineData("Debug", /*aot*/ true, /*singleFileBundle*/ false)]
+    [InlineData("Release", /*aot*/ false, /*singleFileBundle*/ false)]
+    [InlineData("Release", /*aot*/ true, /*singleFileBundle*/ false)]
+    [InlineData("Debug", /*aot*/ false, /*singleFileBundle*/ true)]
+    // [InlineData("Debug", /*aot*/ true, /*singleFileBundle*/ true)] // https://github.com/dotnet/runtime/issues/95273
+    [InlineData("Release", /*aot*/ false, /*singleFileBundle*/ true)]
+    // [InlineData("Release", /*aot*/ true, /*singleFileBundle*/ true)] // https://github.com/dotnet/runtime/issues/95273
+    public void ConsoleBuildThenRunThenPublish(string config, bool aot, bool singleFileBundle)
     {
         string id = $"{config}_{GetRandomId()}";
         string projectFile = CreateWasmTemplateProject(id, "wasiconsole");
         string projectName = Path.GetFileNameWithoutExtension(projectFile);
-        File.WriteAllText(Path.Combine(_projectDir!, "Program.cs"), s_simpleMainWithArgs);
+        File.Copy(Path.Combine(BuildEnvironment.TestAssetsPath, "SimpleMainWithArgs.cs"), Path.Combine(_projectDir!, "Program.cs"), true);
 
-        string extraProperties = "<WasmSingleFileBundle>true</WasmSingleFileBundle>";
-        if (relinking)
-            extraProperties += "<WasmBuildNative>true</WasmBuildNative>";
-        if (invariantTimezone)
-            extraProperties += "<InvariantTimezone>true</InvariantTimezone>";
+        var buildArgs = new BuildArgs(projectName, config, aot, id, null);
+        buildArgs = ExpandBuildArgs(buildArgs);
 
+        string extraProperties = "";
+        if (aot)
+            extraProperties = "<RunAOTCompilation>true</RunAOTCompilation><_WasmDevel>false</_WasmDevel>";
+        if (singleFileBundle)
+            extraProperties += "<WasmSingleFileBundle>true</WasmSingleFileBundle>";
+        if (!string.IsNullOrEmpty(extraProperties))
         AddItemsPropertiesToProject(projectFile, extraProperties);
 
-        var buildArgs = new BuildArgs(projectName, config, /*aot*/false, id, null);
-        buildArgs = ExpandBuildArgs(buildArgs);
-
-        bool expectRelinking = config == "Release" || relinking;
         BuildProject(buildArgs,
                     id: id,
                     new BuildProjectOptions(
-                        DotnetWasmFromRuntimePack: !expectRelinking,
+                        DotnetWasmFromRuntimePack: true,
```

Comment:
This does not look correct. This should be `false` for relinking, and aot. How is it not failing at all?

Reply:
It does not fail, no matter what value we pass to `DotnetWasmFromRuntimePack`. We use `DotnetWasmFromRuntimePack` only in interpreted version, see:
https://github.com/dotnet/runtime/blob/f66c1c15409ff505ba55f8fdf86bc5c041bf47a3/src/mono/wasm/Wasm.Build.Tests/TestMainJsProjectProvider.cs#L89-L93
Never again we check for its value.

Reply:
Hm this should get cleaned up to be same as wasm in an upcoming PR to share more code between wasm-bt and wasi-bt.